deep_qa/training/text_trainer.py

```diff
@@ -64,22 +60,15 @@ class TextTrainer(Trainer):
         LSTMs, if you don't specify an output size in the ``encoder`` params.
     embedding_dropout: float, optional (default=0.5)
         Dropout parameter to apply to the embedding layer
-    use_data_generator: bool, optional (default=False)
-        In :func:`~TextTrainer.create_data_arrays`, should we return one big, padded data array, or
-        should we return a data generator?  Using a data generator allows for dynamic
-        (batch-specific) padding, which can dramatically reduce runtime, but your model must be
-        written correctly to take advantage of that.  See :func:`~TextTrainer._set_padding_lengths`
-        and :func:`~TextTrainer._get_padding_lengths` for more details on that.
-    use_dynamic_padding: bool, optional (default=False)
-        If ``True``, we will set padding lengths based on the data `per batch`, instead of on the
-        whole dataset.  This only works if ``use_data_generator`` is ``True``, if your model is
-        structured to allow variable-length sequences (typically using ``None`` for specific
-        dimensions when you build your model), and if you don't set padding values in
-        :func:`~TextTrainer._set_padding_lengths`.  This flag specifically is read in
-        :func:`~TextTrainer._set_padding_lengths` to know if we should set certain padding values
-        or not.  It's handled correctly for ``num_sentence_words`` and ``num_word_characters`` in
-        this class, but you need to be sure to implement it correctly in subclasses for this to
-        work.
+    data_generator: Dict[str, Any], optional (default=None)
```

Comment:
Can we add a fairly prominent warning here that using this will make your batches non-iid if your labels are correlated with input length and that this problem is compounded with smaller amounts of data.